torch/masked/maskedtensor/binary.py

```diff
@@ -0,0 +1,206 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+
+import torch
+
+from .core import _masks_match, _tensors_match, is_masked_tensor
+
+__all__ = []  # type: ignore[var-annotated]
+
+BINARY_NAMES = [
+    "add",
+    #    "addcdiv",
+    #    "addcmul",
+    "atan2",
+    "arctan2",
+    "bitwise_and",
+    "bitwise_or",
+    "bitwise_xor",
+    "bitwise_left_shift",
+    "bitwise_right_shift",
+    "div",
+    "divide",
+    "floor_divide",
+    "fmod",
+    "logaddexp",
+    "logaddexp2",
+    #    "logical_and",
+    #    "logical_or",
+    #    "logical_xor",
+    "mul",
+    "multiply",
+    "nextafter",
+    "remainder",
+    "sub",
+    "subtract",
+    "true_divide",
+    "eq",
+    "ne",
+    "le",
+    "ge",
+    "greater",
+    "greater_equal",
+    # "equal",
+    "gt",
+    # "isclose",
+    "less_equal",
+    "lt",
+    "less",
+    "maximum",
+    "minimum",
+    "fmax",
+    "fmin",
+    "not_equal",
+]
+
+INPLACE_BINARY_NAMES = [
+    n + "_"
+    for n in (
+        list(
+            set(BINARY_NAMES)
+            - {
+                "logaddexp",
+                "logaddexp2",
+                "equal",
+                "fmin",
+                "minimum",
+                "maximum",
+                "fmax",
+            }
+        )
+    )
+]
+
+
+def _get_at_least_one_mask(a, b):
+    if not is_masked_tensor(a) and not is_masked_tensor(b):
+        raise TypeError("At least one of `a` and `b` must be a MaskedTensor")
+    if not _masks_match(a, b):
+        raise ValueError("a and b must have matching masks")
+    if is_masked_tensor(a):
+        return a.get_mask()
+    return b.get_mask()
+
+
+def _binary_helper(fn, args, kwargs, inplace):
+    from .passthrough import _map_mt_args_kwargs, _wrap_result
+
+    if len(kwargs) != 0:
+        raise ValueError("len(kwargs) must equal 0")
+    if len(args) > 2:
+        for a in args[1:]:
+            if torch.is_tensor(a):
+                raise TypeError("MaskedTensor binary ops do not support Tensor arguments")
```

Comment:
Ahhh good call, I think you're right. I think `args[1:]` should just be `args[2:]` -- that is, we don't support Tensor arguments beyond the LHS and RHS for binary ops. Does that seem more reasonable?

Reply:
It seems more reasonable to me, but you're the authority 😛  if that's the correct invariant, i'd
- comment it
- adjust the error message
- remove the if, since it's redundant with `args[2:]`
(re this last, same could be done for the `if len(args) > 1' / `for args[1:]` in the unary_helper, IIRC

Reply:
Yep that sounds great :)